zebra-chain/src/fmt.rs

```diff
@@ -26,3 +42,27 @@ impl<T> fmt::Debug for SummaryDebug<&Vec<T>> {
         write!(f, "{}, len={}", std::any::type_name::<T>(), self.0.len())
     }
 }
+
+impl<T> ops::Deref for SummaryDebug<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> From<T> for SummaryDebug<T> {
+    fn from(t: T) -> Self {
+        Self(t)
+    }
+}
+
+impl<T> IntoIterator for SummaryDebug<Vec<T>> {
+    type Item = T;
+
+    type IntoIter = <Vec<T> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
```

Comment:
```suggestion
impl<T: IntoIterator> IntoIterator for SummaryDebug<T> {
    type Item = T::Item;

    type IntoIter = T::IntoIter;

    fn into_iter(self) -> Self::IntoIter {
```

Would it make sense to make this more generic?

Reply:
Yes, in fact we can make it generic over any collection, and any exact-size iterator.

See 76c0b074242d717dae56aaf275862fc4c51f4da8